libsplinter/src/error/constraint_violation.rs

```diff
@@ -104,6 +108,43 @@ impl ConstraintViolationError {
         Self {
             violation_type,
             source: Some(source),
+            store: None,
+            operation: None,
+        }
+    }
+
+    /// Constructs a new `ConstraintViolationError` from a specified source error and violation
+    /// type with store and operation.
+    ///
+    /// The implementation of `std::fmt::Display` for this error will  pass through the
+    /// display of the source message prefixed with the store and operation.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use splinter::error::{ ConstraintViolationError, ConstraintViolationType };
+    ///
+    /// let db_err = std::io::Error::new(std::io::ErrorKind::Other, "db error");
+    /// let constraint_violation_error =
+    ///     ConstraintViolationError::from_source_with_violation_type_with_store(
+    ///          ConstraintViolationType::Unique,
+    ///          Box::new(db_err),
+    ///         "DbStore".to_string(),
+    ///         "db_operation".to_string()
+    /// );
+    /// assert_eq!(format!("{}", constraint_violation_error), "DbStore db_operation: db error");
+    /// ```
+    pub fn from_source_with_violation_type_with_store(
```

Comment:
Should this be "from_source_with_violation_type_and_store"? I thought we had an example of using "and" but I can't find it.

Reply:
updated